tools/cli/src/mentat_cli/input.rs

```diff
@@ -118,13 +123,17 @@ impl InputReader {
                         Ok(More)
                     },
                     _ => {
+                        let history = self.buffer.clone();
+                        self.add_history(&history);
```

Comment:
Just make `self.add_history` take a `String`, and pass `self.buffer.clone()` here.